### DISCUSSION TITLE
Setting to increase available history with up arrow

### DISCUSSION THREAD

**Discussion by CharlesARoy** [marlonrichert/zsh-autocomplete#407] [Q&A] [answered]
Just a quick question:

Is there a setting to allow one to cycle through more than the past 16 commands in the history (while still only showing 16 at a time)?

If not, this feature would be appreciated as it is the standard behavior in bash/zsh.

**Comment by marlonrichert** [comment, marked as answer]
Yes, there's a setting for that, documented here:
https://github.com/marlonrichert/zsh-autocomplete/blob/39423112977a8c520962bc11c46ee31e7ca873ca/.zshrc#L25

**Reply by CharlesARoy** [reply]
Thanks for the tip! I tried that setting but the number of lines I set are displayed when I press the up arrow. It also maxes out at the number of lines that can be displayed on my screen, so for example, if I set that parameter to 100, only 48 lines are returned.

What I was hoping for is a setting that would allow me to scroll up through $HISTSIZE number of history entries while only displaying say, 10 at a time. Does that option exist?

**Reply by marlonrichert** [reply]
No, sorry, that's not possible with Zsh technically. If you need more/better tools to inspect (and even edit) your history, have a look at my [Zsh Hist plugin](https://github.com/marlonrichert/zsh-hist).

**Reply by CharlesARoy** [reply]
Thanks again for the tips. I don't want to beat this discussion to death, but just wanted to ask one last clarifying question. When you say "that's not possible with Zsh technically", do you mean that it's not possible to show X number of history entries AND scroll up indefinitely, or just that it's not possible to scroll back through the history indefinitely?

Just asking because in my experience with Zsh, it seems possible to scroll up through the history as much as you want, e.g.:

![2022-01-21 09 10 46](https://user-images.githubusercontent.com/46253001/150570934-bf5aded2-6d91-4dc2-b4dd-bf1bdf6d5b7f.gif)

**Reply by marlonrichert** [reply]
> it's not possible to show X number of history entries AND scroll up indefinitely

This is what I meant. If you want to be able to scroll from the last to the first history item, then the menu will automatically occupy all available lines on screen. It’s not possible to limit it to showing just X lines in that case.
